### ISSUE TITLE
Dimension conventions

### ISSUE THREAD

**Issue by julj** [fchollet/keras#3925]
I am not very clear about dimension conventions. More specifically:

- I found out what 'tf' and 'th' image_dim_ordering conventions were by googling around, but I could not find it in the doc or in the code (keras/backend/common.py). It is (depth, dim1, dim2) for 'th' and (dim1, dim2, depth) for 'tf', right?

- If I got it right, I must set my input shape according to that parameter. So for 1d signal (I mostly work with raw audio), depth=1 and I should have the following shape:
    - 'th' convention: (batchsize, 1, dim1, dim2), with one of the dim set to 1
    - 'tf' convention: (batchsize, dim1, dim2, 1), with one of the dim set to 1

    Now, on which dim is the convolution applied in Convolution1D? I need to know which dim I must set to 1.

**Comment by kgrm** [comment]
>It is (depth, dim1, dim2) for 'th' and (dim1, dim2, depth) for 'tf', right?

Yes.

>Now, on which dim is the convolution applied in Convolution1D? I need to know which dim I must set to 1.

Both backends use the same dim convention for time-series (1-dimensional) convolution: (samples, time_steps, feature_size), and the convolution is applied along the time axis.

**Comment by fchollet** [comment]
The documentation for every layer has an "input shape" section which answers your questions precisely.

https://keras.io/layers/convolutional/

> Convolution2D
> Input shape
> 4D tensor with shape: (samples, channels, rows, cols) if dim_ordering='th' or 4D tensor with shape: (samples, rows, cols, channels) if dim_ordering='tf'.

> Convolution1D
> Input shape
> 3D tensor with shape: (samples, steps, input_dim).
